### ISSUE TITLE
Objects literals with function as values not work

### ISSUE THREAD

**Issue by mtdb** [rzimmerman/kal#80]
for example

``` javascript
1: a = {b:1
2:      c:2}
3: print a.b
```

Works fine, but

``` javascript
1: a = {b:1
2:      c:function()
3:           return 2}
4: print a.c()
```

Not compile, I get this error: **Expected Block on line 2 in file test.kal**

**Comment by rzimmerman** [comment]
I tried various permutations of indentation as well and I get the same error. Thanks for filing this, I'll mark it for the next release (0.4.6).

**Comment by rzimmerman** [comment]
Actually this is turning out to be pretty complicated. The part of the compiler that allows multiline objects squashes any newlines, so your code is turning into:

``` kal
a = {b:1,c:function() return 2}
print(a.c())
```

before it even gets parsed by the grammar. It's going to take some work to fix, so I'm pushing this forward to 0.4.7 so I can get a new release out with a bunch of other important fixes.

**Comment by rzimmerman** [comment]
da575bc ads support for functions in "implicit maps", but no "explicit maps". Implicit means no `{`s. So this will work:

``` kal
a = 
  b: 1
  c: function ()
    return 2
  d: ->
    return 3
```

But this will not:

``` kal
a = {b: 1
  c: function ()
    return 2
  d: ->
    return 3
  }
```
